tests/virt_autotest/host_upgrade_step3_run.pm

```diff
@@ -28,6 +29,19 @@ sub get_script_run {
 
 sub run {
     my $self = shift;
+
+    #Install qa test repo
+    my $installed_product = get_var('VERSION_TO_INSTALL', get_var('VERSION', ''));
+    my $upgrade_product   = get_required_var('UPGRADE_PRODUCT');
+    my ($upgrade_release) = lc($upgrade_product) =~ /sles-([0-9]+-sp[0-9]+)/;
+    if (check_var('ARCH', 'x86_64') && $installed_product =~ /12-sp5/img && $upgrade_product =~ /15-sp2/img) {
+        my $qa_test_repo = 'http://dist.nue.suse.com/ibs/QA:/Head/SLE-' . uc($upgrade_release);
```

Comment:
It is ok to add this as temporary workaround for this specific test scenario. However we need to find out the root cause why this workaround is needed. Is it actually a product bug?

Reply:
qa_lib_virtauto is not any package from SLES product. Tend to workaround this asap firstly. I am not very keen to debug the root cause of this.

Reply:
I can not open the verification job. 
BTW, regarding the two affected tests in acceptance group, which you give in "Related ticket", do you have the test results of them via either manual verification or with this automation workaround? I need to collect the results of them to send acceptance report, which should be given today.

Reply:
1. Updated automation verification run: http://10.67.19.99/tests/721 
You should can access to this. Otherwise, check your network connection.
2. Manual run to confirm the functionality is fine:
http://qadb2.suse.de/qadb/submission.php?submission_id=2317159

Reply:
Finalized implementation here.

Reply:
> qa_lib_virtauto is not any package from SLES product. Tend to workaround this asap firstly. I am not very keen to debug the root cause of this.

If I remember correctly, we already add the upgrade product's QA repo in its autoyast file for offline upgrade, and also add the qa_lib_virtauto pkg in the autoyst's software section to ensure it is installed. Then if it is still missing or incomplete, it looks like a product bug. I am afraid it is QA's responsibility to locate it and fire a bug if confirmed to be an issue.

Reply:
> 1. Updated automation verification run: http://10.67.19.99/tests/721
>    You should can access to this. Otherwise, check your network connection.
> 2. Manual run to confirm the functionality is fine:
>    http://qadb2.suse.de/qadb/submission.php?submission_id=2317159

Thanks for it. I can open them now. Both are for kvm, do you have xen result now? Or I should wait some time for it ?

Reply:
I do not think there is any need for xen verification. It is obvious that they failed due to the same problem.

Reply:
> I do not think there is any need for xen verification. It is obvious that they failed due to the same problem.

It is obvious that they fail for same reason. But it does not mean that when this failure is workarounded, and kvm upgrade test pass, then xen upgrade test should pass. If so, we will not need to "dup" the tests, right?

Reply:
I will merge it now to let the xen test be rerun.

Reply:
> > qa_lib_virtauto is not any package from SLES product. Tend to workaround this asap firstly. I am not very keen to debug the root cause of this.
> 
> If I remember correctly, we already add the upgrade product's QA repo in its autoyast file for offline upgrade, and also add the qa_lib_virtauto pkg in the autoyst's software section to ensure it is installed. Then if it is still missing or incomplete, it looks like a product bug. I am afraid it is QA's responsibility to locate it and fire a bug if confirmed to be an issue.

I do not see your further reply to it. Does it mean you agree?

Reply:
1. Because 80% problem caused by 20% failure reason. Sometimes pragmatic assumption is very useful and righ.
2. If you insist on a xen manual run, I can not guarantee a complete result before the end of today, although I already put full effort into this Beta2 activity.
3. Anyway a xen autotmation run for this pull request is not needed definitely.

Reply:
> 1. Because 80% problem caused by 20% failure reason. Sometimes pragmatic assumption is very useful and right.

I agree. But in milestone testing, we should not rely on assumption. We need real test result.

> 2. If you insist on a xen manual run, I can not guarantee a complete result before the end of today, although I already put full effort into this Beta2 activity.

Thanks for the efforts. I will send acceptance report based on current status. Let's include this xen test result in milestone full report. We can base on the automation test result on osd after retrigger.

> 3. Anyway a xen autotmation run for this pull request is not needed definitely.
Agree, this xen run is not for PR validation, but for real milestone testing.